### PR TITLE
[server] Fix mis-reporting of errors as "success" (in addition)

### PR DESCRIPTION
**Pull request by geropl**
## Description
Noticed while checking metrics, that all errors get reported as "success" as well :grimacing: - the line ended up in the wrong branch. 

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/user/token-service.ts [hunk 1]
```diff
@@ -125,9 +125,10 @@ export class TokenService implements TokenProvider {
                     // Perform actual refresh
                     const stopTimer = scmTokenRefreshLatencyHistogram.startTimer({ host });
                     try {
-                        return await authProvider.refreshToken(user, requestedLifetimeDate);
-                    } finally {
+                        const result = await authProvider.refreshToken(user, requestedLifetimeDate);
```
Comment (kylos101) [inline components/server/src/user/token-service.ts:128]: 👋 should we add a catch, and `reportScmTokenRefreshRequest(host, opportunisticRefresh, "some_failure");`?

Comment (geropl) [reply components/server/src/user/token-service.ts:128]: I agree that feels a bit off. But error cases are handled separately below ([link](https://github.com/gitpod-io/gitpod/pull/19750/files#diff-f2ae81f929de2489e54cc19f07753a11bc982553cc55c9bb60a6f4958e956d9fR139-R154)).

Comment (kylos101) [reply components/server/src/user/token-service.ts:128]: Ah, I see, an outer try/catch block - thank you!

**Review by kylos101 — APPROVED** [review]
:shipit:

**Pull request merged**